lib/models/device.coffee

```diff
@@ -291,16 +227,9 @@ exports.has = (name, callback) ->
 #		console.log("Is device online? #{isOnline}")
 ###
 exports.isOnline = (name, callback) ->
-
-	if not callback?
-		throw new errors.ResinMissingParameter('callback')
-
-	if not _.isFunction(callback)
-		throw new errors.ResinInvalidParameter('callback', callback, 'not a function')
-
-	exports.get name, (error, device) ->
-		return callback(error) if error?
-		return callback(null, !!device.is_online)
+	exports.get(name).then (device) ->
+		return !!device.is_online
```

Comment:
I remember needing this because I was getting `is_online: 1`, `is_online: 0`, as well as `is_online: true`, `is_online: false`. I can remove the `!!` if you confirm that we're using only integers or booleans now.